### PR TITLE
PWN-8284 - Pending transactions for claim and send

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-8284
https://p2pvalidator.atlassian.net/browse/PWN-8397

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/bridge/api/mapper/BridgeMapper.kt [hunk 1]
```diff
@@ -35,7 +36,7 @@ class BridgeMapper {
             transactions = response.transactions.orEmpty(),
             signatures = response.signatures.orEmpty(),
             fees = fromNetwork(response.fees),
-            status = response.status,
+            status = ClaimStatus.IN_PROGRESS,
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/bridge/api/mapper/BridgeMapper.kt:39]: revert please

File: app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt [hunk 2]
```diff
@@ -24,4 +24,10 @@ class HistoryPresenter(
     override fun onSellTransactionClicked(transactionId: String) {
         view?.openSellTransactionDetails(transactionId)
     }
+
+    override fun onClaimPendingClicked(transactionId: String) {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt:28]: ???
maybe = Unit ?

File: app/src/main/res/values/bridges_strings.xml [hunk 3]
```diff
@@ -50,9 +50,10 @@
     <string name="bridge_usdc_usdt_swap_action">Swap USDC → USDCet</string>
     <string name="bridge_how_to_claim_for_free_title">How to claim for free</string>
     <string name="bridge_claim_description_progress">It usually takes 15–20 minutes for a transaction to complete</string>
-    <string name="bridge_ethereum_network">Ethereum network</string>
-    <string name="bridge_claim_pending">Claiming…</string>
-    <string name="bridge_send_pending">Sending…</string>
+    <string name="bridge_to_ethereum">To Ethereum</string>
```
Comment (kabramovp2p) [inline app/src/main/res/values/bridges_strings.xml:53]: i've already added this
transaction_history_wh_send

Comment (kabramovp2p) [reply app/src/main/res/values/bridges_strings.xml:53]: done in PWN-8397 - use title and subtitle from designs
like others
check commit pls

Comment (kabramovp2p) [reply app/src/main/res/values/bridges_strings.xml:53]: i mean we need to remove duplicates
my string or your
and use it from one source

File: app/src/main/java/org/p2p/wallet/bridge/api/mapper/BridgeMapper.kt [hunk 4]
```diff
@@ -94,7 +94,7 @@ class BridgeMapper {
             recipient = response.recipient,
             amount = fromNetwork(response.amount),
             fees = fromNetwork(response.fees),
-            status = fromNetwork(response.status),
+            status = BridgeSendTransactionStatus.IN_PROGRESS,
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/bridge/api/mapper/BridgeMapper.kt:97]: revert

File: app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt [hunk 5]
```diff
@@ -24,4 +39,57 @@ class HistoryPresenter(
...
+            view?.showProgressDialog(
+                bundleId = bridgeBundle.bundleId,
+                progressDetails = progressDetails
+            )
+        }
+    }
+
+    override fun onSendPendingClicked(transactionId: String) {
+        launch {
+            val sendBundle = ethereumInteractor.getSendBundleById(transactionId) ?: return@launch
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/history/HistoryPresenter.kt:74]: No need for try?

File: app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryPresenter.kt [hunk 6]
```diff
@@ -43,6 +56,59 @@ class TokenHistoryPresenter(
         view?.openSellTransactionDetails(transactionId)
     }
 
+    override fun onBridgePendingClaimClicked(transactionId: String) {
+        launch {
+            val bridgeBundle = ethereumInteractor.getClaimBundleById(transactionId) ?: return@launch
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/token/TokenHistoryPresenter.kt:61]: I Suggest to extract to separate class to remove duplication. Something like BridgeHistoryItemClickHandler

File: app/src/main/java/org/p2p/wallet/history/ui/token/adapter/HistoryAdapter.kt [hunk 7]
```diff
@@ -86,6 +86,8 @@ class HistoryAdapter(
             is DateItem -> item.date.hashCode().toLong()
             is UserSendLinksItem -> item.linksCount.hashCode().toLong()
             is HistoryItem.SwapBannerItem -> item.date.hashCode().toLong()
+            is HistoryItem.BridgeClaimItem -> item.transactionId.toLong()
+            is HistoryItem.BridgeSendItem -> item.transactionId.toLong()
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/token/adapter/HistoryAdapter.kt:90]: Transaction ID is Long? Wow

File: app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/BridgePendingViewHolder.kt [hunk 8]
```diff
@@ -19,14 +19,15 @@ class BridgePendingViewHolder(
...
         itemView.setOnClickListener { onHistoryClicked(item) }
         with(binding) {
-            // TODO fix icons
-            transactionTokenImageView.apply {
-                ERC20Tokens.values()
-                    .firstOrNull { it.mintAddress == item.sendDetails.recipient.raw }
-                    ?.let { setTokenImage(glideManager, it.tokenIconUrl) }
-                    ?: setTransactionIcon(R.drawable.ic_transaction_send)
+            val foundToken = ERC20Tokens.values()
+                .firstOrNull { it.replaceTokenSymbol == item.sendDetails.amount.symbol }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/BridgePendingViewHolder.kt:23]: Again, we need a method for searching token with different params

File: app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt [hunk 9]
```diff
@@ -86,7 +86,8 @@ class UserTokensPolling(
                         delay(POLLING_ETH_DELAY.inWholeMilliseconds)
                         joinAll(
                             async { fetchSolTokens() },
-                            async { fetchEthereumTokens() }
+                            async { fetchEthereumTokens() },
+                            // TODO - move this code to separate worker
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/home/ui/main/UserTokensPolling.kt:90]: Task

**Review by gslevinkov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/BridgePendingViewHolder.kt (hunk 8 above)
Comment (Davran331122) [reply app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/BridgePendingViewHolder.kt:23]: Yeah, we can do it. But for now I think it's useless

Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/history/ui/token/adapter/holders/BridgePendingViewHolder.kt:23]: Now or never

also values is a performance eating operation
https://stackoverflow.com/questions/2446135/is-there-a-performance-hit-when-using-enum-values-vs-string-arrays

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**